openapi_python_client/parser/properties/union.py

```diff
@@ -80,22 +80,44 @@ def build(
         sub_properties: list[PropertyProtocol] = []
 
         type_list_data = []
-        if isinstance(data.type, list):
+        if isinstance(data.type, list) and not (data.anyOf or data.oneOf):
```

Comment:
This is for a simple case where the schema is just like:
```
MyType:
  type: ["string", "integer"]
```

That's equivalent in meaning to `anyOf: [{type: string}, {type: int}]`, so on lines 84-85 we convert it into those explicit variant types. However, if there was redundantly an `anyOf` or `oneOf` _and_ a multiple `type:`, then we do not want to do that, because the variants that are described in the `anyOf`/`oneOf` list already fully describe the possible types. For instance:

```yaml
MyType:
  type: ["string", "integer"]
  oneOf:
    - type: string
       format: date-time
    - type: integer
       default: 3
```

In that example, prior to my fix, the union would've ended up having four variants instead of two.